### PR TITLE
feat: multi speech credential diff labels but same vendor

### PR DESCRIPTION
**Pull request by xquanluu**
#424

### CODE REVIEW

File: lib/session/call-session.js [hunk 1]
```diff
@@ -640,14 +659,16 @@ class CallSession extends Emitter {
...
     const {writeAlerts, AlertType} = this.srf.locals;
     if (this.accountInfo.speech && this.accountInfo.speech.length > 0) {
-      const credential = this.accountInfo.speech.find((s) => s.vendor === vendor);
+      const credential = this.accountInfo.speech.find((s) => s.vendor === vendor &&
+        ((label && s.label === label) || label === null));
       if (credential && (
         (type === 'tts' && credential.use_for_tts) ||
         (type === 'stt' && credential.use_for_stt)
       )) {
+        this.logger.info(`Speech credential vendor: ${credential.vendor} and Label: ${credential.label} is chosen`);
```
Comment (davehorton) [inline lib/session/call-session.js:671]: I'd prefer the info message not to mention label if label is null

File: lib/tasks/config.js [hunk 2]
```diff
@@ -105,6 +105,9 @@ class TaskConfig extends Task {
       cs.speechSynthesisVendor = this.synthesizer.vendor !== 'default'
         ? this.synthesizer.vendor
         : cs.speechSynthesisVendor;
+      cs.speechSynthesisLabel = this.synthesizer.label !== 'default'
```
Comment (davehorton) [inline lib/tasks/config.js:108]: why would you compare label to 'default' ?  That doesn't make sense, there is no default label

File: lib/tasks/config.js [hunk 3]
```diff
@@ -117,6 +120,9 @@ class TaskConfig extends Task {
       cs.speechRecognizerVendor = this.recognizer.vendor !== 'default'
         ? this.recognizer.vendor
         : cs.speechRecognizerVendor;
+      cs.speechRecognizerLabel = this.recognizer.label !== 'default'
+        ? this.recognizer.label
```
Comment (davehorton) [inline lib/tasks/config.js:124]: same comment about default

**Review by davehorton — CHANGES_REQUESTED** [review]

File: lib/session/call-session.js (hunk 1 above)
Comment (xquanluu) [reply lib/session/call-session.js:671]: fixed

File: lib/tasks/config.js (hunk 2 above)
Comment (xquanluu) [reply lib/tasks/config.js:108]: I am having same question like you, for example vendor and language are having same behavior. why is it like that?

I think for config verb, when a parameter = default means, the configure verb will use the value provided by application and not overide the parameter value in that call.

File: lib/tasks/config.js (hunk 3 above)
Comment (xquanluu) [reply lib/tasks/config.js:124]: same as above answer

File: lib/tasks/config.js (hunk 2 above)
Comment (davehorton) [reply lib/tasks/config.js:108]: the reason for vendor and language is that after setting applications defaults, someone might want to change something like hints with config or say but not vendor or language.  I suppose that you are right....the same could be true of label.  So I will close this

**Review by davehorton — APPROVED** [review]

**Pull request merged**